src/modules/user/user.usecase.ts

```diff
@@ -808,4 +812,123 @@ export class UserUseCases {
 
     return this.avatarService.getDownloadUrl(avatarKey);
   }
+
+  async changeUserEmailById(userUuid: string, newEmail: string) {
+    const user = await this.userRepository.findByUuid(userUuid);
+
+    if (!user) {
+      throw new UserNotFoundError();
+    }
+
+    const maybeAlreadyExistentUser =
+      await this.userRepository.findByUsername(newEmail);
+
+    const userAlreadyExists = !!maybeAlreadyExistentUser;
+
+    if (userAlreadyExists) {
+      throw new UserAlreadyRegisteredError(newEmail);
+    }
+
+    const { uuid, email } = user;
+
+    await this.userRepository.updateById(user.id, {
+      email: newEmail,
+      username: newEmail,
+      bridgeUser: newEmail,
+    });
+    await this.sharedWorkspaceRepository.updateGuestEmail(email, newEmail);
+
+    this.networkService.updateUserEmail(uuid, newEmail).catch(async () => {
```

Comment:
The deal was: 
1. Update the network email
2. Update the drive email references

If 2 fails -> do the rollback of 1. 

But here I see 2 before 1.